src/Homepage/index.js

```diff
@@ -1,197 +1,206 @@
-    return item
-      ? {
-          item: searchResultToFormEntry(this.state.searchResult)
-        }
-      : undefined;
-  }
+    getLinkState() {
+        const item = searchResultToFormEntry(this.state.searchResult);
+        return item
+            ? {
+                item: searchResultToFormEntry(this.state.searchResult)
+            }
+            : undefined;
+    }
 
-  getLinkTo() {
-    const state = this.getLinkState();
-    return state && state["Store Name"]
-      ? { pathname: "/update", state }
-      : { pathname: "/location" };
-  }
+    getLinkTo() {
+        const state = this.getLinkState();
+        return state && state["Store Name"]
+            ? {pathname: "/update", state}
+            : {pathname: "/location"};
+    }
 
-  render() {
-    let missingBlock = null;
-    if (this.isMissingLocationInformation(this.state.searchResult)) {
-      const { searchResult } = this.state;
-      missingBlock = (
-        <MissingBlock
-          missing={true}
-          result={{
-            name: searchResult.name,
-            entries: [searchResultToFormEntry(searchResult)]
-          }}
-        ></MissingBlock>
-      );
+    render() {
+        let missingBlock = null;
+        if (this.isMissingLocationInformation(this.state.searchResult)) {
+            const {searchResult} = this.state;
+            missingBlock = (
+                <MissingBlock
+                    missing={true}
+                    result={{
+                        name: searchResult.name,
+                        entries: [searchResultToFormEntry(searchResult)]
+                    }}
+                ></MissingBlock>
+            );
+        }
+        const closeByResults = this.state.results.filter(
+            result => result.distance < 200000
+        );
+        return (
+            <div>
+                <MapWithSearch
+                    value=""
+                    onSuccess={result => {
+                        if (result && result.latLng) {
+                            this.setState(
+                                {
+                                    searchResultLatlng: result.latLng,
+                                    searchResult: result
+                                },
+                                this.onBoundsChanged(result.latLng)
+                            );
+                        }
+                    }}
+                    style={{height: "85vh", width: "100vw"}}
+                    position={this.state.searchResultLatlng}
+                    locations={this.state.markers}
+                    onBoundsChanged={center => this.onBoundsChanged(center)}
+                    onPositionChanged={position =>
+                        this.setState({
+                            searchResultLatlng: position,
+                            searchResult: null
+                        })
+                    }
+                />
+                <Accordion className="map-wide-data-container">
```

Comment:
1. support marker clicks: if a marker is clicked, the corresponding card should be visible.
2. support "add first entry for a store": if user searches for a store, which is not found, we create a `missingBlock` for them to easily add new info for the store. when this happens, the accordion should show up.